code/game/machinery/scan_gate.dm

```diff
@@ -162,7 +189,17 @@
 		beep = !beep
 	if(beep)
 		alarm_beep()
+		if(!ignore_signals)
+			color = wires.get_color_of_wire(WIRE_ACCEPT)
+			var/obj/item/assembly/S = wires.get_attached(color)
+			if(istype(S))
+				S.activate()
 	else
+		if(!ignore_signals)
+			color = wires.get_color_of_wire(WIRE_DENY)
+			var/obj/item/assembly/S = wires.get_attached(color)
+			if(istype(S))
```

Comment:
Is it possible for this to not be an assembly, other than it being null? Because if not, just do `S?.activate()`.